MediaGalleryUi/Model/Listing/DataProvider.php

```diff
@@ -84,4 +100,29 @@ public function getData(): array
             ];
         }
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function addFilter(Filter $filter)
+    {
+        if ($filter->getConditionType() === self::FULLTEXT_CONDITION_TYPE) {
+            $this->fulltextFilter = $filter;
+        } else {
+            parent::addFilter($filter);
+        }
+    }
+
+    /**
+     * Add fulltext filter to SearchCriteriaBuilder and reset SearchCriteria
+     *
+     * @return void
+     */
+    private function prepareSearchCriteria(): void
+    {
+        if ($this->fulltextFilter) {
+            $this->searchCriteria = null;
```

Comment:
What is the reason of clearing search criteria and custom flow for the fulltext filter?

Reply:
so it breaks Select